openmc/statepoint.py

```diff
@@ -94,475 +91,385 @@ def __init__(self, filename):
-                elif '-yn' in score:
-                    score = score.replace('-yn', '-' + str(moments[j]))
+    @property
+    def n_particles(self):
+        return self._f['n_particles'].value
 
-                tally.add_score(score)
+    @property
+    def n_realizations(self):
+        return self._f['n_realizations'].value
 
-            # Add Tally to the global dictionary of all Tallies
-            self.tallies[tally_key] = tally
+    @property
+    def path(self):
+        return self._f['path'].value.decode()
 
-    def read_results(self):
-        """Read tally results and store them in the ``tallies`` attribute. No results
-        are read when the statepoint is instantiated.
+    @property
+    def run_mode(self):
+        return self._f['run_mode'].value.decode()
 
-        """
+    @property
+    def seed(self):
+        return self._f['seed'].value
 
-        # Number of realizations for global Tallies
-        self._n_realizations = self._f['n_realizations'].value
+    @property
+    def source(self):
+        if self.source_present:
+            return self._f['source_bank'].value
+        else:
+            return None
 
-        # Read global Tallies
-        n_global_tallies = self._f['n_global_tallies'].value
+    @property
+    def source_present(self):
+        return self._f['source_present'].value > 0
 
-        data = self._f['global_tallies'].value
-        self._global_tallies = np.column_stack((data['sum'], data['sum_sq']))
+    @property
+    def tallies(self):
+        if not self._tallies_read:
+            # Initialize dictionary for tallies
+            self._tallies = {}
 
-        # Flag indicating if Tallies are present
-        self._tallies_present = self._f['tallies/tallies_present'].value
+            # Read the number of tallies
+            n_tallies = self._f['tallies/n_tallies'].value
 
-        base = 'tallies/tally '
+            # Read a list of the IDs for each Tally
+            if n_tallies > 0:
+                # OpenMC Tally IDs (redefined internally from user definitions)
+                tally_keys = self._f['tallies/keys'].value
+            else:
+                tally_keys = []
 
-        # Read Tally results
-        if self._tallies_present:
+            base = 'tallies/tally '
 
-            # Iterate over and extract the results for all Tallies
-            for tally_key in self._tally_keys:
+            # Iterate over all Tallies
+            for tally_key in tally_keys:
 
-                # Get this Tally
-                tally = self._tallies[tally_key]
+                # Read the Tally size specifications
+                n_realizations = self._f['{0}{1}/n_realizations'.format(base, tally_key)].value
 
-                # Compute the total number of bins for this Tally
-                num_tot_bins = tally.num_bins
+                # Create Tally object and assign basic properties
+                tally = openmc.Tally(tally_key)
```

Comment:
Perhaps add the ``id=tally_key`` keyword argument here just to be perfectly clear that the tally keys are the user-specified tally IDs rather than user-specified tally names.